Task 17

Fodrigoal opened the pull request:
I hope that's what you were asking for.

pmvald commented on tasks/task-017/button-menu.html:
```diff
@@ -0,0 +1,49 @@
+                <p>Calendar</p>
+            </button>
+        </li>
+        <!-- added 3 more icons -->
```
good!

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,31 @@
+    position: relative;
+}
+
+.s-btn-menu,
+button {
```
something is wrong here! If by this style you mean the button that is inside `s-btn-menu` then you should not use comma, just space! Also since we are using scss this button style should be inside the previous block of s-btn-menu.

Fodrigoal replied:
ok!

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,31 @@
+    padding: 1em;
+}
+
+.s-btn-menu,
+button:hover {
```
same as above

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,31 @@
+    color: #1E74FD;
+}
+
+.s-btn-menu,
+i {
```
same as above

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,31 @@
+    font-size: 1.5em;
+}
+
+.s-btn-menu,
+p {
```
same as above

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,31 @@
+.s-btn-menu,
+button {
+    color: black;
+    padding: 1em;
```
for spacing between elements use margin (see the design-guide.md)

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,31 @@
+.s-btn-menu,
+p {
+    font-size: 1em;
+    padding-top: 0.5em;
```
for spacing between elements use margin (see the design-guide.md)

Fodrigoal replied:
Of course! My mistake.

pmvald reviewed (changes requested):
@Fodrigoal I reviewed the code. There was a problem with using scss syntax. Please review scss syntax and follow my comments. thanks.

Fodrigoal commented:
Hi @pmvald I've done some changes to the scss.

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,19 @@
+.s-btn-menu {
+    letter-spacing: 0;
```
no need

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,19 @@
+.s-btn-menu {
+    letter-spacing: 0;
+    height: 56px;
```
write all the spaces in terms of $spacer. Note that $spacer=16px. So this could be `4.5*$spacer`

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,19 @@
+.s-btn-menu {
+    letter-spacing: 0;
+    height: 56px;
+    line-height: 1.2em;
```
As mentioned in the previous review, we use *margin* for elements spacing, not padding, line-height, etc.

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,19 @@
+    display: flex;
+    position: relative;
+    button {
+        color: black;
```
No need for color. color will inherit from the parent (keeping the design minimal)

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,19 @@
+    button {
+        color: black;
+        &:hover {
+            color: #1E74FD;
```
no need. We don't have hover in mobile devices

pmvald commented on -sass/styles/_menus.m.scss:
```diff
@@ -0,0 +1,19 @@
+        font-size: 1.5em;
+    }
+    p {
+        margin-top: 0.5em;
```
use $spacer for spacing

pmvald reviewed (changes requested):
@Fodrigoal Better now. Please attend to the comments I made

Fodrigoal commented:
I've noticed display (same as menu from _menus.scss) and position weren't necessary so I deleted them.

I have also deleted all the spacing because the layout still looks good and similar to Mobilekit without them.

pmvald commented:
@Fodrigoal I made some changes and merged the branch. Please take a look at my changes to get a better idea of the design system. Yes, in general we try to be minimalist and use classes/style as least as possible. In this case, the `<menu>` styles already do most of things for us. For menu, we always have to style the "active" (=selected) item as well (look at my code). I also changed the button color to "inherit". Your follow-up assignment: what is inherit style, and why I changed the color to inherit?
-

Fodrigoal commented:
Inherit is a keyword that specifies that a property should inherit its value from its parent element. You used it so the button color would emulate the color from its parent (li). Otherwise, the button would change to blue due to a rule for button tags on the _reboot.scss file.

Fodrigoal commented:
Would you mind explaining to me why we need to style the active item in the menu and why didn't you change the <i> font size and put it nested on the button?

Something like this:
```
.s-btn-menu {

 
    li > button {
        color: inherit;

        & > i {
            font-size: $font-size-sm*2;
        }

        & > p {
            font-size: $font-size-sm;
        }

        &.active {
            color: $primary-color;
        }
    }
}
```

Thanks @pmvald
